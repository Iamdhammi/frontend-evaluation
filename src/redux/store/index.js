import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState = {};

const middleware = [thunk];

// create Store
const store = createStore(
    rootReducer, // all combined reducers
    initialState, // DUHH initaial state at craete store level
    compose(
        applyMiddleware(...middleware)
    )
);

export default store;