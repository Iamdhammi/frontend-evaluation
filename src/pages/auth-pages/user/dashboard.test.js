import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../redux/store';
import Dashboard from './dashboard';
import { Router } from "react-router-dom";
import history from '../../../history'

afterEach(cleanup);

it("loads user dashboard", () => {
    const { asFragment } = render(
        <Provider store={store}>
            <Router history={history}>
                <Dashboard />
            </Router>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});
