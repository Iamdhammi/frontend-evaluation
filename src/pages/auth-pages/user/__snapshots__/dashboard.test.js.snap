// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads user dashboard 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="nav"
    >
      <div
        class="nav__container"
      >
        <div
          class="nav__logo"
        >
          <img
            alt="Logo"
            class="nav__logo--img"
            src="logo.png"
          />
        </div>
        <div
          class="nav_routes"
        >
          <button
            class="ui button"
          >
            Log out
          </button>
        </div>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="Toastify"
      />
      <div
        class="container__header"
      >
        <div>
          <h2>
            My Stories
          </h2>
        </div>
        <div>
          <button
            class="ui button"
          >
            Create story
          </button>
        </div>
      </div>
      <div>
        <div />
      </div>
      <div
        class="container__body"
      >
        <div>
          <div
            class="sc-fznxsB fWHyCt"
          >
            <div
              class="sc-fznWqX jEDPQU"
            >
              <div
                class="sc-fznJRM mmMWz"
              >
                <div
                  style="padding: 24px; margin-top: 100px;"
                >
                  <div
                    class="ui active transition visible inverted dimmer"
                    style="display: flex;"
                  >
                    <div
                      class="content"
                    >
                      <div
                        class="ui inverted text loader"
                      >
                        Loading
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="sc-AxjAm DlzgR rdt_Table"
                role="table"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
